src/stream.rs

```diff
@@ -174,94 +278,96 @@ impl error::Error for StreamError {
     }
 }
 
-/// Extensions to `cpal::Device`
-pub(crate) trait CpalDeviceExt {
-    fn new_output_stream_with_format(
-        &self,
-        format: cpal::SupportedStreamConfig,
-    ) -> Result<(Arc<DynamicMixerController<f32>>, cpal::Stream), cpal::BuildStreamError>;
-
-    fn try_new_output_stream_config(
-        &self,
-        config: cpal::SupportedStreamConfig,
-    ) -> Result<(Arc<DynamicMixerController<f32>>, cpal::Stream), StreamError>;
-}
-
-impl CpalDeviceExt for cpal::Device {
-    fn new_output_stream_with_format(
-        &self,
-        format: cpal::SupportedStreamConfig,
-    ) -> Result<(Arc<DynamicMixerController<f32>>, cpal::Stream), cpal::BuildStreamError> {
-        let (mixer_tx, mut mixer_rx) =
-            dynamic_mixer::mixer::<f32>(format.channels(), format.sample_rate().0);
+impl OutputStream {
+    pub fn open(
+        device: &cpal::Device,
+        config: &OutputStreamConfig,
+    ) -> Result<OutputStream, StreamError> {
+        let (controller, source) = mixer(config.channel_count, config.sample_rate.0);
```

Comment:
> Output stream does not have configurable output type. It could be any Source actually, but here we fixing it to be a Mixer only. 

I would think the most commen usecase for this is to play multiple sources one after the other. That should be easy to do. So maybe this should not take a mixer but a queue like structure?

Reply:
Right now I'd prefer either mixer or some configurable stub, since at the moment those are the simplest options. 

Maybe queue-like behavior can be extracted from `Sink` for re-use. Or as I mentioned before my preference would be to split Sink into simplest functions (queue-like behavior would be one of them) and let users only add ones they need. E.g. if I were designing the library I'd made functions like `crossfade::crossfade(self, other, duration)` public. As a bonus user-provided filers can be configured the same way as the embedded ones. Currently some functions are in `impl Sink` and users cannot add their stuff there.

Reply:
I think in games sounds are normally played in response to some events, so they should not be strictly ordered.

Reply:
You've convinced me :+1: